File: task03/go/subtask3.go
package main

import (
    "fmt"
    "os"
    "strconv"
    "bufio"
)

func main() {
    fmt.Print("Enter a number: ")
    scanner := bufio.NewScanner(os.Stdin)
    scanner.Scan()
    n, _ := strconv.Atoi(scanner.Text())

    for i := 0; i < n; i++ {
        for j := 0; j < n - i - 1; j++ {
            fmt.Print(" ")
        }
        for j := 0; j < 2 * i + 1; j++ {
            fmt.Print("*")
        }
        fmt.Println()
    }

    for i := n - 2; i >= 0; i-- {
        for j := 0; j < n - i - 1; j++ {
            fmt.Print(" ")
        }
        for j := 0; j < 2 * i + 1; j++ {
            fmt.Print("*")
        }
        fmt.Println()
    }
}

File: task03/c++/subtask3.cpp
#include <iostream>

int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            std::cout << " ";
        }
        for (int j = 0; j < 2 * i + 1; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }

    for (int i = n - 2; i >= 0; i--) {
        for (int j = 0; j < n - i - 1; j++) {
            std::cout << " ";
        }
        for (int j = 0; j < 2 * i + 1; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }

    return 0;
}